### PR TITLE
[DS-153] Feature/tooltip

### PR DESCRIPTION
**Pull request by panvourtsis**
## Description

This PR introduce a new component for our system, a tooltip. 

This tooltip is only holding plain text. You can define position but based on the screen this will change if there is no space to show it on the pre-defined location.

A new system config name `overrides` was introduced. Its mainly for tooltip ATM in order to define its color and shade for the whole system. The logic is that tooltip will have only one color that will be unique for the whole project. This way we may introduce new overrides in the future.

[Ticket](image)
[Design](https://www.figma.com/file/8kMPBNYHHXz2AtkzeeDmk5/Design-System?node-id=659%3A13)

Note: Asset Matching was changed due to the fact we were using `tippyjs` for the tooltip and now i change it.

## Screenshot

image
image

## Test Plan

* storyshots

### CODE REVIEW

**Review by whosyourtaco — APPROVED** [review]
LGTM 🎉

**Comment by whosyourtaco** [comment]
Only thing I would comment on is the sizes of the tooltip, because `normal` and `medium` sound kind of the same to me, but they represent different sizes.

File: src/index.tsx [hunk 1]
```diff
@@ -63,5 +66,7 @@ export {
   Palette,
   Typography,
   Elevation,
+  overrides,
```
Comment (Christos-Zacharopoulos) [inline src/index.tsx:69]: Do we need to expose this?

File: src/components/AssetMatching/components/Asset/AssetExternalLink.tsx [hunk 2]
```diff
@@ -1,29 +1,20 @@
...
 const AssetExternalLink: FC<ExternalLinkProps> = ({ tooltipContent, url }) => {
   return (
-    <Tippy
-      content={tooltipContent}
-      arrow
-      placement={'right'}
-      allowHTML={false}
-      css={Styles.tooltip}
-    >
+    <Tooltip content={tooltipContent} placement={'right'}>
```
Comment (michalisKout) [inline src/components/AssetMatching/components/Asset/AssetExternalLink.tsx:13]: before: 
Screenshot 2021-06-17 at 10 20 29 AM
after:
Screenshot 2021-06-17 at 10 20 39 AM

You should also pass the `Styles.tooltip` to the new tooltip component.

File: src/index.tsx (hunk 1 above)
Comment (panvourtsis) [reply src/index.tsx:69]: in case we want to know overrides, i didnt had a case in mind but we expose all others for the same reasoning

**Review by michalisKout — CHANGES_REQUESTED** [review]

File: src/components/Tooltip/Tooltip.style.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    if (size === 'large') {
+      return theme.typography.fontSizes['16'];
+    } else if (size === 'small') {
+      return theme.typography.fontSizes['12'];
+    }
+
+    return theme.typography.fontSizes['14'];
+  };
+
+  return css`
```
Comment (michalisKout) [inline src/components/Tooltip/Tooltip.style.ts:24]: you can avoid all the `!important` by adding a new parent to the tooltip. Example:
jsx
```
   <div css={tooltipStyle({ size, placement })}>
        <ReactTooltip id={tooltipID} place={placement} effect="solid" className="tooltip">
          {content}
        </ReactTooltip>
      </div>
```

styles
```
   > div {
      color: ${pickTextColorFromSwatches(color, shade)};
      background-color: ${backgroundColor};
      max-width: ${rem(256)};
      padding: ${theme.spacing.sm};
      font-size: ${defineFontSizeBasedOnTooltipSize(size)};
      line-height: 110%;
      border-radius: ${theme.spacing.sm};
      text-align: justify;
    }

    .show {
      opacity: 1 !important;
    }
    ....
```

File: src/components/Tooltip/Tooltip.style.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    max-width: ${rem(256)};
+    padding: ${theme.spacing.sm} !important;
+    font-size: ${defineFontSizeBasedOnTooltipSize(size)} !important;
+    line-height: 110%;
+    border-radius: ${theme.spacing.sm};
+    text-align: justify;
+    &.show {
+      opacity: 1 !important;
+    }
+    &.place-right {
```
Comment (michalisKout) [inline src/components/Tooltip/Tooltip.style.ts:36]: you can make this more dynamic

example:
```
const getPlacement = (backgroundColor: string, placement?: string) =>
  placement &&
  css`
    .place-${placement} {
      ::after {
        border-${placement}-color: ${backgroundColor} !important;
      }
    }
  `;
```

usage
${getPlacement(placement, backgroundColor)}

File: src/components/Tooltip/Tooltip.style.ts (hunk 3 above)
Comment (michalisKout) [reply src/components/Tooltip/Tooltip.style.ts:24]: You can also replace  `> div` with `> ${tooltipId}` to make it more secure for future changes 😄

File: src/components/AssetMatching/components/Asset/AssetExternalLink.tsx (hunk 2 above)
Comment (panvourtsis) [reply src/components/AssetMatching/components/Asset/AssetExternalLink.tsx:13]: dont worry about this, its configurable via theme provider. I already talked with elisavet and i would reach to you for the config needed

File: src/components/Tooltip/Tooltip.style.ts (hunk 3 above)
Comment (panvourtsis) [reply src/components/Tooltip/Tooltip.style.ts:24]: i though of this but is it really !important ? (see what i did there?)

Comment (michalisKout) [reply src/components/Tooltip/Tooltip.style.ts:28]: I think it's always !important to avoid `!important;`

**Pull request merged**